migration/src/main/resources/migrations/ensure-orders-have-encounters.xml

```diff
@@ -17,28 +17,43 @@
            then create the encounters for orders that don’t have encounters
         -->
         <sql>
-            There are over 300,000 orders on both Rwink and Butaro with no linked encounter
-            select count(*) from orders where encounter_id is null
+            SET @EncTUUID :=  "a9ef19d1-8875-11ea-8c74-7a7919290ad6";
 
-            Of these, there are still between 100k and 150k on each server with no encounters on the same day
 
-            select count(*)
-            from orders o
-            left join encounter e on o.patient_id = e.patient_id and date(o.start_date) = date(e.encounter_datetime)
-            where e.encounter_id is null;
+            /* Create EncounterType */
+            INSERT INTO `encounter_type`
+            (`name`,`description`,`creator`,`date_created`,`uuid`)
+            VALUES
+            (
+            "Drug Order Encounter","for orders without encounters",1,now(),@EncTUUID);
 
-            So we need to figure out a strategy for this.
+            /* Get encounterType id */
+            select @encTId:=encounter_type_id from encounter_type where uuid=@EncTUUID;
 
-            Safest thing for now might be to create new encounters for all of these.  Later on, if we want to merge
-            some of these into other existing enocunters for the same patient on the same day, we can do so,
-            but that isn't needed for migration and might not be 100% accurate.
 
-            This could be something like a SQL script that runs the following statements:
+            /* Create encounters */
+            INSERT INTO `encounter`(`encounter_type`,`patient_id`,`encounter_datetime`,`creator`,`date_created`,`voided`,`voided_by`,`date_voided`,`void_reason`,`uuid`)
+            SELECT @encTId,patient_id,start_date,creator,date_created,voided,voided_by,date_voided,void_reason,UUID() FROM orders WHERE encounter_id is null;
+
```

Comment:
A few comments on the above "create encounters" query

1 - can you clean up the formatting a bit so things are nicely aligned (maybe it is just the view I have in github, but if we can improve formatting that would be good)

2 - if I'm reading this right, this will create a new encounter for _every_ order.  So if a patient is prescribed 3 drugs at the same time, this creates 3 encounters?  This might be what I was originally indicated on the ticket to do, but it doesn't seem like what we want to me.  Certainly I'd say if orders have the same matching patient_id, start_date, creator, and date_created associated with them, then they are probably all part of the same encounter and we should group them together as such.  That likely means having your select clause above do a group by on these fields, so that you only create 1 encounter per patient_id, start_date, date_created, creator group.  You may also want to group on only the "date" of date_created, rather than the timestamp.

Reply:
2. don't  you think orders can share the same encounter(same creator,date_created,patient_id) but with different start date?

Reply:
@bjoshuadevelopment yes that probably makes sense.  I think what you have done looks good.